src/_pytest/_code/source.py

```diff
@@ -337,7 +337,10 @@ def get_statement_startend2(lineno, node):
 
 def getstatementrange_ast(lineno, source, assertion=False, astnode=None):
     if astnode is None:
-        content = str(source)
+        if six.PY2:
+            content = unicode(source)
```

Comment:
Unfortunately this still might fail, because `unicode` will try to use the `ascii` codec to decode the bytestream (see py27 failure). Not sure if by this point `getstatementrange_ast` shouldn't already be a unicode stream actually. 🤔